src/main/java/com/github/kornilova_l/flamegraph/plugin/server/converters/file/jmc/FlightRecorderCompressedConverter.java

```diff
@@ -0,0 +1,87 @@
+package com.github.kornilova_l.flamegraph.plugin.server.converters.file.jmc;
+
+import com.github.kornilova_l.flamegraph.plugin.server.converters.file.CFlamegraph;
+import com.github.kornilova_l.flamegraph.plugin.server.converters.file.CFlamegraphLine;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+class FlightRecorderCompressedConverter extends FlightRecorderConverter {
+    private CFlamegraph cFlamegraph;
+
+    public FlightRecorderCompressedConverter(File file) {
+        super(file);
+
+        parseAsCFlamegraph();
+    }
+
+    CFlamegraph getCFlamegraph() {
+        return cFlamegraph;
+    }
+
+    private void parseAsCFlamegraph() {
+        ArrayList<CFlamegraphLine> cFlamegraphLines = new ArrayList<>();
+        HashMap<String, Integer> classNames = new HashMap<>();
+        HashMap<String, Integer> methodNames = new HashMap<>();
+        HashMap<String, Integer> descriptions = new HashMap<>();
+
+        Map<String, Integer> stacks = this.getStacks();
+        Set<String> stackFrames = stacks.keySet();
+
+        for(String frame : stackFrames) {
+            Integer width = stacks.get(frame);
+
+            String[] calls = frame.split(";");
+            Integer depth = calls.length;
+            String topFrame = calls[calls.length - 1]; //TODO: Add a check on length before this access
+            String className = getClassName(topFrame);
+            String methodName = getMethodName(topFrame);
+
+            CFlamegraphLine line = new CFlamegraphLine(
+                    getId(classNames, className),
+                    getId(methodNames, methodName),
+                    null,
+                    width,
+                    depth);
+            cFlamegraphLines.add(line);
+        }
+
+        this.cFlamegraph = new CFlamegraph(cFlamegraphLines, toArray(classNames), toArray(methodNames), toArray(descriptions));
+    }
+
+    private Integer getId(HashMap<String, Integer> map, String name) {
+        Integer id = map.get(name);
+        if (id == null) {
+            Integer newId = map.size();
+            map.put(name, newId);
+            return newId;
+        }
+        return id;
+    }
+
+    private String[] toArray(HashMap<String, Integer> names)  {
+        return names.keySet().toArray(new String[0]);
+    }
+
+    private String getClassName(String call) {
```

Comment:
Use `StacksToTreeBuilder.getClassName` etc instead

Reply:
Actually `StacksToTreeBuilder` should be removed because it's only used to construct tree from flamegraph stacks. So just move needed methods to `ProfilerToCompressedFlamegraphConverter`

Reply:
It's also used in the Yourkit Converter class. We'd have to find a better class to put this in.
![image](https://user-images.githubusercontent.com/38335968/49036386-75ce0c80-f16c-11e8-86f6-42a6c4987b60.png)

`StacksToTreeBuilder.getClassName` is also a little different than more simplistic `FlightRecorderCompressedConverter.getClassName` method, which has only a single string parameter. If we consolidate the parsing, perhaps we should change the former method so it doesn't rely on character positions in the method signature?

Reply:
`StacksToTreeBuilder.getClassName` is so weird because it's optimized for parsing huge flamegraph files (and yourkit csv files), so we do not scan a string twice while looking for a `.` or space character, so let's leave it as it is :)